features/netsocket/nsapi_dns.cpp

```diff
@@ -194,24 +292,145 @@ static int dns_scan_response(const uint8_t **p, nsapi_addr_t *addr, unsigned add
     return count;
 }
 
+static void nsapi_dns_cache_add(const char *host, nsapi_addr_t *address, uint32_t ttl)
+{
+    // RFC 1034: if TTL is zero, entry is not added to cache
+    if (!ttl) {
```

Comment:
As this is a number, not a boolean, I prefer `if (ttl == 0)`

Reply:
Corrected.